src/controllers/observation-controller.ts

```diff
@@ -54,60 +62,81 @@ export default class ObservationController extends AbstractController {
     this.router.put('/:id', this.editObservation);
     this.router.delete('/:id', this.removeObservation);
     this.router.post('/set-verification', this.setVerificationStatus);
+
     return this.router;
   }
 
   private getObservations = async (req: RequestWithPageParams, res: Response, next: NextFunction): Promise<void> => {
     try {
-      const { lang = 'eng' }: { lang: string } = req.query;
-      const langOrigin = LocaleOrigin[lang] ? LocaleOrigin[lang] : 'desc_eng';
-
-      const paramsSearch = parsePageParams(req.query);
-      const paramsAggregation = parseWhereParams(req.query, req.user);
-      const observations = await this.observations.findAndCount(Object.assign(paramsSearch, paramsAggregation));
-
-      const content = observations[0].map(obs => {
-        // sanitaze user's data
-        const finder = Object.assign({}, obs.finder.sanitizeUser(), { id: obs.finder.id });
+      const { query, user } = req;
+      const { lang = 'eng' }: { lang: string } = query;
+      const langOrigin = LocaleOrigin[lang] || 'desc_eng';
+
+      const paramsSearch = parsePageParams(query);
+      const paramsAggregation = parseWhereParams(user, query);
+      const [observations, totalElements] = await this.observations.findAndCount(
+        Object.assign(paramsSearch, paramsAggregation),
+      );
+
+      const content = observations.map(observation => {
+        // sanitize user's data
+        const finder = Object.assign({}, User.sanitizeUser(observation.finder));
         // transform 'observation'
-        const observation = Object.entries(obs)
-          // clear 'filter' field
-          .filter(([ObservationField]) => ObservationField !== 'ring')
+        const observationEntries = Object.entries(observation)
+          // remove 'ring' field
+          .filter(([field]) => field !== 'ring')
           // map 'lang' param according 'Locale'
-          .map(([ObservationKey, ObservationValue]) => {
-            if (typeof ObservationValue === 'object' && ObservationValue !== null) {
-              const value = Object.entries(ObservationValue)
-                .filter(([subfield]) => filterFieldByLocale(subfield as Locale, langOrigin))
-                .map(([subfield, subValue]) =>
-                  subfield === langOrigin ? ['desc', subValue] : ([subfield, subValue] as any),
-                )
-                .reduce((acc, [subfield, subValue]) => Object.assign(acc, { [subfield]: subValue }), {});
-              return [ObservationKey, value];
-            }
-            return [ObservationKey, ObservationValue];
-          })
-          .reduce(
-            (acc, [ObservationField, ObservationValue]) => Object.assign(acc, { [ObservationField]: ObservationValue }),
-            {},
-          );
-        return Object.assign({}, observation, { finder }, { ring: obs.ring.id });
+          .map(entry => mapLocale(entry, langOrigin))
```

Comment:
`mapLocale` should be unaware about exact field to choose, I mean this method should decide this internally and never inside calling it methods

Reply:
This is kind of broken encapsulation which in current case will cause code duplication in future

Reply:
yeah, got it. Now 'query' is passed to mapLocale as the second argument.